# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit/BaselineFrame.cpp
# Commit: 8c9daa76d87a
# Full Hash: 8c9daa76d87aef8bf071bef81d9990ccd211983b
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-10-17 21:54:25
# Description:
#   Bug 1586762 - Set BaselineFrame::interpreterICEntry correctly for BaselineDebugModeOSR at debug prologue. r=iain
#   
#   We were using the first pc, but that skips the prologue's type monitor ICs.
#   
#   This bug is pretty hard to trigger in practice (I was unable to write a test
# ==============================================================================

diff -r e641756cd54f -r 8c9daa76d87a js/src/jit/BaselineFrame.cpp
--- a/js/src/jit/BaselineFrame.cpp	Thu Oct 17 15:29:05 2019 +0000
+++ b/js/src/jit/BaselineFrame.cpp	Thu Oct 17 15:33:48 2019 +0000
@@ -119,7 +119,7 @@
   interpreterICEntry_ = jitScript->interpreterICEntryFromPCOffset(pcOffset);
 }
 
-void BaselineFrame::setInterpreterFieldsForPrologueBailout(JSScript* script) {
+void BaselineFrame::setInterpreterFieldsForPrologue(JSScript* script) {
   JitScript* jitScript = script->jitScript();
   interpreterScript_ = script;
   interpreterPC_ = script->code();